test/test_interfaces.py

```diff
@@ -57,7 +71,95 @@ def test_examples_not_valid_path(self):
         with self.assertRaises(FileNotFoundError):
             interface = Interface(lambda x: x, "textbox", "label", examples='invalid-path')
             interface.launch()
+            
+    def test_test_launch(self):
+        with captured_output() as (out, err):
+            prediction_fn = lambda x: x
+            prediction_fn.__name__ = "prediction_fn"
+            interface = Interface(prediction_fn, "textbox", "label")
+            interface.test_launch()
+            output = out.getvalue().strip()
+            self.assertEqual(output, 'Test launch: prediction_fn()... PASSED')
+            
+    @mock.patch("time.sleep")
+    def test_run_until_interupted(self, mock_sleep):
+        with self.assertRaises(KeyboardInterrupt):
+            with captured_output() as (out, err):
+                mock_sleep.side_effect = KeyboardInterrupt()
+                interface = Interface(lambda x: x, "textbox", "label")
+                interface.enable_queue = False
+                thread = threading.Thread()
+                thread.keep_running = mock.MagicMock()
+                interface.run_until_interrupted(thread, None)
+                output = out.getvalue().strip()
+                self.assertEqual(output, 'Keyboard interruption in main thread... closing server.')
+
 
+    @mock.patch('gradio.utils.colab_check')
```

Comment:
It looks like this actually generates a share url? We should just be mindful of not generating too many share urls in our tests, because each time we test, it will use up ports on the gradio api server. I think 1 or 2 is fine, though. Some suggestions:
* Set `analytics_enabled` to `False` so it doesn't affect our analytics. In fact, we should be doing this throughout our tests, and we aren't, so I'll patch this throughout the codebase.
*  Make sure to close the Interface at the end of the test

Reply:
added interface.close to any test that launches the interface